pkg/controller/apigatewayv2/deployment/zz_controller.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package deployment
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	svcapi "github.com/aws/aws-sdk-go/service/apigatewayv2"
+	svcsdkapi "github.com/aws/aws-sdk-go/service/apigatewayv2/apigatewayv2iface"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	cpresource "github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/apigatewayv2/v1alpha1"
+	awsclient "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+const (
+	errUnexpectedObject = "managed resource is not an repository resource"
+
+	errCreateSession = "cannot create a new session"
+	errCreate        = "cannot create Deployment in AWS"
+	errDescribe      = "failed to describe Deployment"
+	errDelete        = "failed to delete Deployment"
+)
+
+type connector struct {
+	kube client.Client
+}
+
+func (c *connector) Connect(ctx context.Context, mg cpresource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*svcapitypes.Deployment)
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
+	}
+	cfg, err := awsclient.GetConfigV1(ctx, c.kube, mg, cr.Spec.ForProvider.Region)
+	if err != nil {
+		return nil, err
+	}
+	sess, err := session.NewSession(cfg)
```

Comment:
looks like this line is reused across `Connect` methods, does it make sense to move to `GetConfigV1`?